src/Plug.vala

```diff
@@ -367,10 +369,39 @@ namespace Power {
                 grid.attach (dim_label, 0, 0, 1, 1);
                 grid.attach (dim_switch, 1, 0, 1, 1);
             }
-
             return grid;
         }
 
+        private static bool lid_detect () {
+            var lid_path = File.new_for_path ("/proc/acpi/button/lid/");
+            var enumerator = lid_path.enumerate_children ("standard::*", 0);
```

Comment:
As you don't use any attribute afterward, I would suggest that you only request GLib.FileAttribute.STANDARD_NAME so that all the other attributes are not populated (it speeds up the process)

Reply:
Should I use FileQueryInfoFlags.NONE instead of 0 or better to keep 0?

Thanks a lot!

Reply:
Yeah, better be explicit so that we know why there is a zero there
Thanks for your work :)